src/transformers/safetensors_conversion.py

```diff
@@ -0,0 +1,99 @@
+from typing import Optional
+
+from huggingface_hub import Discussion, HfApi
+
+from .utils import cached_file, logging
+
+
+logger = logging.get_logger(__name__)
+
+
+def previous_pr(api: "HfApi", model_id: str, pr_title: str) -> Optional["Discussion"]:
+    try:
+        main_commit = api.list_repo_commits(model_id)[0].commit_id
+        discussions = api.get_repo_discussions(repo_id=model_id)
+    except Exception:
+        return None
+    for discussion in discussions:
+        if discussion.status == "open" and discussion.is_pull_request and discussion.title == pr_title:
+            commits = api.list_repo_commits(model_id, revision=discussion.git_reference)
+
+            if main_commit == commits[1].commit_id:
+                return discussion
+    return None
```

Comment:
we can implem a Hub API to replace this whole function IMO cc @SBrandeis

Reply:
Courtesy of @SBrandeis (https://github.com/huggingface/huggingface_hub/pull/1845):

```py
for discussion in get_repo_discussions(
    repo_id="openai/whisper-large-v3",
    author="sanchit-gandhi",
    discussion_type="pull_request",
    discussion_status="open",
):
    ...
```

Reply:
Thanks Simon! Applied in [d32a18e](https://github.com/huggingface/transformers/pull/27571/commits/d32a18e46b4827293d85ad2f6f2a8cd5a4e9a362)